doc/guide/cockpit-session.xml

```diff
@@ -58,6 +58,32 @@ promise.then(user => { ... });
     </variablelist>
 
     <para>Returns a promise that completes once the user information is available.</para>
+    <warning>
+	    <para>
+		    <code>cockpit.user()</code> is soft-deprecated since Cockpit 336, <link
+		    linkend="cockpit-init">cockpit.init()</link> can be used
+		    instead to obtain the user on initial page load.
+
+		    If your page needs to maintain compatibility with
+		    older Cockpit versions you can fallback to
+		    <code>cockpit.user()</code> like the example below:
+<programlisting>
+try {
+  await cockpit.init();
+} catch (exp) {
+  if (exp.problem === 'not-supported') {
+    const user = await cockpit.user();
+    cockpit.info.user = {
```

Comment:
Do we really want to recommend that to people?  I know we do it in cockpit-files, but ... yikes.

Reply:
(not a veto, of course... but maybe we should make it clear that this is a hack or at least a polyfill)

Reply:
So hear me out :) 

The reason I send this PR is that I think we aren't doing an amazing job at advocating our sparse API users to move over to the new hotness. So advertising this in our documentation is a good idea.

I would be happy to not include this snippet, but it seems useful? So I am happy to enhance this with a "HACK", or clear note that it is a polyfill.

Reply:
We could also document that to use this feature you need to have `"requires": { "cockpit": "337" }` in the manifest?

Reply:
Also, we expect cockpit.init() to be called at the very top level, *once*. It's not meant to be called multiple times somewhere in the code, so this should point out that fact.

Reply:
I have reworded this to recommend using cockpit.info if you don't need to maintain compatibility with versions older then 336. I think that should be clear enough?

Documenting requires sounds good but also a bit "random" here, same goes cockpit.info(), that information should be in the `cockpit.info` section.